src/core/config/mac.h

```diff
@@ -399,14 +399,27 @@
 /**
  * @def OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD
  *
- * For some reasons, CSL receivers wake up a little later than expected. This variable specifies how much time that
- * CSL receiver would wake up earlier than the expected sample window. The time is in unit of 10 symbols.
+ * Reception scheduling and ramp up time needed for the receiver to be ready, in unit of 10 symbols.
  *
  */
 #ifndef OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD
 #define OPENTHREAD_CONFIG_CSL_RECEIVE_TIME_AHEAD 2
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_CSL_MIN_RECEIVE_ON
+ *
+ * The total duration, in units of microseconds, required for the CSL receiver to fully receive and acknowledge a frame
+ * of maximum possible length.
+ * - Maximum frame size with preamble: 6*2+127*2 symbols
+ * - AIFS: 12 symbols
+ * - Maximum ACK size with preamble: 6*2+33*2 symbols
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_CSL_MIN_RECEIVE_ON
+#define OPENTHREAD_CONFIG_CSL_MIN_RECEIVE_ON 356 * 16
```

Comment:
I don't think we need to assume the worse case. We can keep the sample window just big enough to receive the preamble of packet. `otPlatRadioSleep` just let radio driver knows it is OK to sleep. But when the preamble of frame is received, and radio is in the middle of receiving a frame, it will wait until the end of the frame then goes to sleep.

We have made changes in PR #5449 and #5504 to make sure radio only sleep when it is truly idle and the sleep API will not break any ongoing transmission or reception(`nrf_802154_sleep` --> `nrf_802154_sleep_if_idle`).

Reply:
> But when the preamble of frame is received, and radio is in the middle of receiving a frame, it will wait until the end of the frame then goes to sleep

`radio.h` does not define `otPlatRadioSleep()` like this. I think it is up to given platform port to abort receiving and enter sleep mode, or continue receiving until the frame is received. Until `otPlatRadioSleep()` clearly states that it shall not abort ongoing reception, I think it is needed to assume worst case.

Reply:
I agree. Looks like we have to update the API description in `radio.h` or create a new API called `otPlatRadioSleepIfIdle` for this purpose.

Reply:
Yes, I was commenting the same, your assumptions @brianljt are only for nRF platform. This configuration option allows for every platform to adapt for it's requirements. Optimized values can be configured for different platforms.

Reply:
One more thing. With this worst case assumption, should `kMinCslWindow` be `OPENTHREAD_CONFIG_CSL_MIN_RECEIVE_ON * 2` as CSL phase is in the center of the sample window? This creates a minimum sample window bigger than 11ms which could make CSL less appealing than data poll regarding power consumption.

Reply:
You are right about adding the _max_frame_ to the second half of the sample window.

Also about the low performance for the worst case, but in real products it can be greatly improved, specially if the device's temperature is taken into account for the crystal's PPM value.